### PR TITLE
update to work with Windows and helm chart improvements

### PR DESCRIPTION
**Pull request by mikenorgate**

### CODE REVIEW

File: nginx/config/base.go [hunk 1]
```diff
@@ -17,8 +17,13 @@ package config
 const BaseTemplate = `
 worker_processes 4;
 worker_rlimit_nofile 4096;
+{{if .is_windows}}
+pid C:/tmp/nginx.pid;
+{{else}}
 pid /tmp/nginx.pid;
+{{end}}
 user root root;
```
Comment (colwill) [inline nginx/config/base.go:25]: does `user root root` work on windows? Does this require adding a root account in addition to the Administrator account

File: nginx/config/base.go [hunk 2]
```diff
@@ -17,8 +17,13 @@ package config
...
+error_log stderr;
 
 events {
   worker_connections 2048;
```
Comment (colwill) [inline nginx/config/base.go:29]: Why such low connections?

File: utils/netutil/netutil.go [hunk 3]
```diff
@@ -17,12 +17,13 @@ import (
...
 	"fmt"
 	"net"
+	"strings"
 	"time"
 )
 
 // _supportedInterfaces is an ordered list of ip interfaces from which
 // host ip is determined.
-var _supportedInterfaces = []string{"eth0", "ib0"}
+var _supportedInterfaces = []string{"eth0", "ib0", "ethernet", "Ethernet", "vEthernet", "vethernet"}
```
Comment (colwill) [inline utils/netutil/netutil.go:26]: Does this also need `vEthernet (Default Switch)` just in case?

**Review by colwill — APPROVED** [review]

File: nginx/config/base.go (hunk 1 above)
Comment (mikenorgate) [reply nginx/config/base.go:25]: No, Windows ignores the user setting

File: nginx/config/base.go (hunk 2 above)
Comment (mikenorgate) [reply nginx/config/base.go:29]: This was the default config it came with, it shouldn't really ever handle that many concurrent http requests anyway

File: utils/netutil/netutil.go (hunk 3 above)
Comment (mikenorgate) [reply utils/netutil/netutil.go:26]: No, if you look a few lines down I trim the interface name at the first space. I found that the "name" bit varied depending on where is was running so was best to ignore it

**Pull request merged**